tests/e2e/sample-applications/mysql-persistent/mysql-persistent-template.yaml

```diff
@@ -0,0 +1,151 @@
+apiVersion: v1
+kind: List
+items:
+  - kind: Namespace
+    apiVersion: v1
+    metadata:
+      name: mysql-persistent
+      labels:
+        app: mysql
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: mysql-persistent-sa
+      namespace: mysql-persistent
+      labels:
+        component: mysql-persistent
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: mysql
+      namespace: mysql-persistent
+      labels:
+        app: mysql
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      storageClassName: gp2
+      resources:
+        requests:
+          storage: 10Gi
+  - kind: SecurityContextConstraints
+    apiVersion: security.openshift.io/v1
+    metadata:
+      name: mysql-persistent-scc
+    allowPrivilegeEscalation: true
+    allowPrivilegedContainer: true
+    runAsUser:
+      type: RunAsAny
+    seLinuxContext:
+      type: RunAsAny
+    fsGroup:
+      type: RunAsAny
+    supplementalGroups:
+      type: RunAsAny
+    volumes:
+    - '*'
+    users:
+    - system:admin
+    - system:serviceaccount:mysql-persistent:mysql-persistent-sa
+  - apiVersion: v1
+    kind: Secret
+    metadata:
+      name: mysql
+      namespace: mysql-persistent
+      labels:
+        app: mysql
+    stringData:
+      database-name: MYSQL_DATABASE
+      database-password: MYSQL_PASSWORD
+      database-root-password: MYSQL_ROOT_PASSWORD
+      database-user: MYSQL_USER
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      annotations:
+        template.openshift.io/expose-uri: mysql://{.spec.clusterIP}:{.spec.ports[?(.name=="mysql")].port}
+      name: mysql
+      namespace: mysql-persistent
+      labels:
+        app: mysql
+    spec:
+      ports:
+      - name: mysql
+        port: 3306
+      selector:
+        name: mysql
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: mysql
+      namespace: mysql-persistent
+      labels:
+        app: mysql
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      storageClassName: gp2
```

Comment:
The gp2 storageClass is specific to AWS, and the deployment fails due to pending pvc for clusters deployed outside aws. Maybe this could be left blank, so that we use the clsuter's default storage class?

Reply:
yes, agree. The MSSQL had the storageclasses hardcoded, so made it in a similar way, cause I didn't want to change the logic. I plan later to refactor the apps, but not as part of this commit cause it's already too big.